src/tray.js

```diff
@@ -96,11 +122,19 @@ exports.create = function(config) {
             }
         }
 
-        trayIcon.setImage(newFavicon);
+        trayIcon && trayIcon.setImage(newFavicon);
```

Comment:
Won't this now interfere with the blinking?

Reply:
Correct, added a condition to check if it's already blinking. In that case we don't set the icon explicitly instead only changing the stored icon that's used in the blink function. It's not fatal if we replace the icon but it's definitely a non-polished UX (the icon can reappear too early with the new badge if it's currently hidden during the blink loop).